TypeCobol.LanguageServer/Workspace.cs

```diff
@@ -951,6 +953,203 @@ private void FinalCompilationStepCompleted(object cUnit, ProgramClassEvent progr
+                if (IsIgnored(dataDefinition))
+                {
+                    // Ignore this node
+                    return null;
+                }
+
+                var row = new StringBuilder();
+
+                //TODO manage slack bytes (property is dataDefinition.SlackBytes)
+
+                // Line number (starting at 1)
+                AppendToRow(dataDefinition.CodeElement.GetLineInMainSource() + 1);
+
+                /// Level number
+                AppendToRow(dataDefinition.CodeElement.LevelNumber);
+
+                // Name (preceded by an indent depending on the level)
+                AppendToRow(GetIndent(dataDefinition) + GetName(dataDefinition));
+
+                // Declaration (Picture, Usage, REDEFINES, OCCURS, ...)
+                AppendToRow(GetDeclaration(dataDefinition));
+
+                // Start/End/Length
+                var start = dataDefinition.StartPosition;
+                var length = dataDefinition.PhysicalLength;
+                AppendToRow(start);
+                AppendToRow(GetEnd(start, length));
+                AppendToRow(length);
+
+                return row.ToString();
+
+                static bool IsIgnored(DataDefinition dataDefinition)
+                {
+                    DataDefinitionEntry codeElement = dataDefinition.CodeElement;
+                    return codeElement == null || codeElement.Line < 0;
+                }
+
+                void AppendToRow(object value)
+                {
+                    row.Append(value).Append(separator);
+                }
+
+                static string GetIndent(DataDefinition dataDefinition)
+                {
+                    var result = new StringBuilder();
+                    Node parentData = dataDefinition.Parent;
+                    while (parentData is not DataSection)
+                    {
+                        parentData = parentData.Parent;
+                        result.Append(INDENT);
+                    }
+
+                    return result.ToString();
+                }
+
+                static string GetName(DataDefinition dataDefinition)
+                {
+                    return dataDefinition.Name == null ? FILLER : dataDefinition.Name;
+                }
+
+                static string GetDeclaration(DataDefinition dataDefinition)
+                {
+                    string result = String.Empty;
+
+                    if (dataDefinition is DataRedefines dataRedefines)
+                    {
+                        result = string.Format(REDEFINES, dataRedefines.CodeElement.RedefinesDataName.Name);
+                    }
+                    else if (dataDefinition is DataDescription dataDescription && dataDescription.CodeElement is DataDescriptionEntry dataDescriptionEntry)
+                    {
+                        if (dataDescription.Picture != null)
+                        {
+                            result = string.Format(PIC, dataDescription.Picture.Value);
```

Comment:
You should use `PictureValidationResult` from `DataDefinition`. `Picture` refers to the text written in source while `PictureValidationResult` is the meaning of it. It will ensure the picture is normalizeed (i.e. `PIC XX` will be returned as `PIC X(2)`)

Reply:
We can keep the text written by the user for the first version of this request.